src/background/messaging/rpc-methods/send-transfer.ts

```diff
@@ -1,36 +1,54 @@
 import { RpcErrorCode, SendTransferRequest } from '@btckit/types';
 
 import { RouteUrls } from '@shared/route-urls';
+import {
+  getRpcParamErrors,
+  rpcSendTransferValidator,
+  validateRpcParams,
+} from '@shared/rpc/methods/validation';
 import { makeRpcErrorResponse } from '@shared/rpc/rpc-methods';
+import { isDefined } from '@shared/utils';
 
 import {
+  RequestParams,
   getTabIdFromPort,
   listenForPopupClose,
   makeSearchParamsWithDefaults,
   triggerRequestWindowOpen,
 } from '../messaging-utils';
 
 export async function rpcSendTransfer(message: SendTransferRequest, port: chrome.runtime.Port) {
-  if (!message.params || !message.params.address || !message.params.amount) {
+  if (!message.params || !validateRpcParams(message.params, rpcSendTransferValidator)) {
+    const errors = getRpcParamErrors(message.params, rpcSendTransferValidator);
     chrome.tabs.sendMessage(
       getTabIdFromPort(port),
       makeRpcErrorResponse('sendTransfer', {
         id: message.id,
         error: {
           code: RpcErrorCode.INVALID_PARAMS,
           message:
-            'Invalid parameters. See the btckit spec for more information: https://btckit.org/docs/spec',
+            'Invalid parameters: ' +
+            errors.map(e => `Error in path ${e.path}, ${e.message}.`).join(' '),
         },
       })
     );
     return;
   }
-  const { urlParams, tabId } = makeSearchParamsWithDefaults(port, [
+
+  const requestParams: RequestParams = [
     ['address', message.params.address],
     ['amount', message.params.amount],
     ['requestId', message.id],
-  ]);
+  ];
+
```

Comment:
Maybe we can add a type for `message` ?

Reply:
`message` is a type here, `SendTransferRequest`, what do you mean here?